types/index.d.ts

```diff
@@ -0,0 +1,431 @@
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import * as Bluebird from "bluebird";
+
+declare function omise(options: Omise.Options): Omise.OmiseStatic;
+
+declare module Omise {
+  interface Options {
+    publicKey: string;
+    secretKey: string;
+  }
+
+  interface OmiseStatic {
+    accounts: Account.Account;
+    balances: Balance.Balance;
+    charges: Charges.Charges;
+    customers: Customers.Customers;
+    disputes: Disputes.Disputes;
+    events: Events.Events;
+    links: Links.Links;
+    recipients: Recipients.Recipients;
+    tokens: Tokens.Tokens;
+    transactions: Transactions.Transactions;
+    transfers: Transfers.Transfers;
+  }
+
+  namespace Account {
+    interface Account {
+      retrieve(callback?: ResponseCallback): Bluebird<Account>;
+    }
+
+    interface Account extends BaseResponse {
+      email: string;
+      currency: string;
+      supported_currencies: [string];
+      created: string;
+    }
+  }
+
+  namespace Balance {
+    interface Balance {
+      retrieve(callback?: ResponseCallback): Bluebird<Balance>;
+    }
+
+    interface Balance extends BaseResponse {
+      available: number;
+      total: number;
+      currency: string;
+    }
+  }
+
+  namespace Cards {
+    interface Card extends BaseResponse {
+      country: string;
+      city: string;
+      postal_code: string;
+      financing: string;
+      bank: string;
+      last_digits: string;
+      brand: string;
+      expiration_month: number;
+      expiration_year: number;
+      fingerprint: string;
+      name: string;
+      security_code_check: boolean;
+      created: string;
+    }
+
+    interface CardRequest {
+      name?: string;
+      expiration_month?: number;
+      expiration_year?: number;
+      postal_code?: string;
+      city?: string;
+    }
+
+    interface CardList extends Pagination.Response {
+      data: [Card];
+    }
+  }
+
+  namespace Charges {
+    interface Charges {
+      create(req: Request, callback?: ResponseCallback): Bluebird<Charge>;
+      update(chargeID: string, req: Request, callback?: ResponseCallback): Bluebird<Charge>;
+      retrieve(chargeID: string, callback?: ResponseCallback): Bluebird<Charge>;
+      list(parameters?: Pagination.Request, callback?: ResponseCallback): Bluebird<ChargeList>;
+      capture(chargeID: string, callback?: ResponseCallback): Bluebird<Charge>;
+      reverse(chargeID: string, callback?: ResponseCallback): Bluebird<Charge>;
+      createRefund(chargeID: string, callback?: ResponseCallback): Bluebird<RefundResponse>;
+      listRefunds(chargeID: string, callback?: ResponseCallback): Bluebird<ListRefundResponse>;
+      retrieveRefund(chargeID: string, refundID: string, callback?: ResponseCallback): Bluebird<RefundResponse>;
+    }
+
+    interface Request {
+      description?: string;
+      amount: number;
+      currency: string;
+      capture?: boolean;
+      card?: string;
+      customer?: string;
+      return_uri?: string;
```

Comment:
`metadata` is missing na
https://www.omise.co/store-your-data-on-omise-with-the-metadata-api

Reply:
Fixed.